ceph/rbd/workflows/namespace.py

```diff
@@ -63,3 +63,67 @@ def create_namespace_and_verify(**kw):
     if namespace_name in ns_ls_out[0]:
         log.info(f"namespace {namespace_name} creation successfully verified")
         return 0
+    else:
+        return 1
+
+
+def remove_namespace_and_verify(**kw):
+    """
+    Removes a namespace in a given pool
+        Args:
+        kw(dict): Key/value pairs that needs to be provided to the installer
+            Example::
+            Supported keys:
+                pool-name(str) : name of the pool in which namespace should be removed, default pool is rbd
+                namespace(str): namespace to removed in the pool
+                See rbd help create for more supported keys
+    """
+    rbd = Rbd(kw["client"])
+
+    # verify if the pool exists, if pool is not passed then consider default pool name i.e rbd
+    pool_name = kw.get("pool-name", "rbd")
+    log.info(f"verifying if pool {pool_name} exists")
+    pool_init_kw = {}
+    pool_init_kw["pool-name"] = pool_name
+    (_, pool_stats_err) = rbd.pool.stats(**pool_init_kw)
+
+    # if pool does not exists,  exit
+    if pool_stats_err:
+        log.error(
+            f"The pool {pool_name} where namespace is suppose to be delete does not exist"
+        )
+        return 1
+
+    # remove a namespace
+    namespace_name = kw.get("namespace", None)
+    log.info(f"namespace {namespace_name} in {pool_name} is getting deleted")
+    ns_delete_kw = {}
+    if namespace_name is None:
+        log.error("namespace name is a must for delete namespace method")
+        return 1
+    _pool_name = kw.get("pool-name", None)
+    ns_delete_kw["namespace"] = namespace_name
+    ns_delete_kw["pool-name"] = _pool_name
+    (_, ns_delete_err) = rbd.namespace.remove(**ns_delete_kw)
+    if not ns_delete_err:
+        log.info(
+            f"SUCCESS: Namespace {namespace_name} got deleted in pool {pool_name} "
+        )
+    else:
+        log.error(
+            f"FAIL: Namespace {namespace_name} deletion failed in pool {pool_name} with error {ns_delete_err}"
+        )
+        return 1
+
+    # verify deleted namespace using namespace ls command
+    log.info(
+        f"deleted namespace {namespace_name} in pool {pool_name} is getting verified"
+    )
+    ns_verify_kw = {}
+    ns_verify_kw["pool-name"] = _pool_name
+    ns_ls_out = rbd.namespace.list(**ns_verify_kw)
+    if namespace_name not in ns_ls_out[0]:
+        log.info(f"namespace {namespace_name} deletion successfully verified")
+        return 0
+    else:
```

Comment:
please add `log.error` if namespace still not get deleted.

Reply:
done